lib/infrastructure/completions/completions_repository.dart

```diff
@@ -54,9 +54,27 @@ class CompletionsRepository extends ICompletionsRepository {
   @override
   Future<void> markAsIncomplete({
     String completionId,
+    bool isResponsePossible,
   }) async {
     try {
-      await _completionsCollection.doc(completionId).delete();
+      if (isResponsePossible) {
+        final LocalUser user = getIt<LocalUser>();
+        await _completionsCollection.doc(completionId).delete();
+        QuerySnapshot responsesSubCollection = await _completionsCollection
+            .doc(completionId)
+            .collection('responses')
+            .where('user_id', isEqualTo: user.id)
+            .get();
+        responsesSubCollection.docs.forEach((element) async {
```

Comment:
iirc, forEach does not work well with async await. We might have to switch to a normal for loop

Reply:
we may not notice it here, but add a print statement after the loop and one within the loop and see which prints first